flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/compiler/NFACompiler.java

```diff
@@ -150,6 +160,59 @@ long getWindowTime() {
 		}
 
 		/**
+		 * Check pattern after match skip strategy.
+		 */
+		private void checkPatternSkipStrategy() {
+			AfterMatchSkipStrategy afterMatchSkipStrategy = currentPattern.getAfterMatchSkipStrategy();
+			if (afterMatchSkipStrategy.getStrategy() == AfterMatchSkipStrategy.SkipStrategy.SKIP_TO_FIRST ||
+				afterMatchSkipStrategy.getStrategy() == AfterMatchSkipStrategy.SkipStrategy.SKIP_TO_LAST) {
+				Pattern<T, ?> pattern = currentPattern;
+				while (!pattern.getName().equals(afterMatchSkipStrategy.getPatternName())) {
+					pattern = pattern.getPrevious();
+				}
+				// pattern name match check.
+				if (pattern == null) {
+					throw new MalformedPatternException("the pattern name specified in AfterMatchSkipStrategy " +
+						"can not be found in the given Pattern");
+				} else {
+					// can not be used with optional states.
+					if (pattern.getQuantifier().hasProperty(Quantifier.QuantifierProperty.OPTIONAL)) {
+						throw new MalformedPatternException("the AfterMatchSkipStrategy "
+							+ afterMatchSkipStrategy.getStrategy() + " can not be used with optional pattern");
+					}
+				}
+
+				// start position check.
+				if (pattern.getPrevious() == null) {
```

Comment:
I personally see no reason for a semantic with RuntimeException. I can't think of any use-case for it. Maybe let's finish this PR without the switch and exceptions and open a JIRA with the switch, ideally with some use-case's for that semantic, so we can further agree on that and see if anyone needs it.